larq_compute_engine/mlir/transforms/optimize.cc

```diff
@@ -58,12 +68,84 @@ DenseElementsAttr Bitpack(PatternRewriter& builder, Attribute x) {
 
 #include "larq_compute_engine/mlir/transforms/generated_optimize.inc"
 
+struct SetBconvReadWriteBitpacked : public OpRewritePattern<TF::LceBconv2dOp> {
+  using OpRewritePattern<TF::LceBconv2dOp>::OpRewritePattern;
+
+  PatternMatchResult matchAndRewrite(TF::LceBconv2dOp bconv_op,
+                                     PatternRewriter& rewriter) const override {
+    Value bconv_input = bconv_op.input();
+    if (!bconv_input.hasOneUse()) return matchFailure();
+
+    auto bconv_input_op =
+        dyn_cast_or_null<TF::LceBconv2dOp>(bconv_input.getDefiningOp());
+    if (!bconv_input_op) return matchFailure();
+
+    if (bconv_input_op.write_bitpacked_output() &&
+        bconv_op.read_bitpacked_input())
+      return matchFailure();
+
+    if (bconv_input_op.padding() == "SAME" && bconv_input_op.pad_values() == 0)
```

Comment:
```suggestion
    if (bconv_input_op.padding() == "SAME" && bconv_input_op.pad_values() != 1)
```

Reply:
I think this is correct because we're checking in order to return `matchFailure`.

Reply:
I agree, this is correct. But for the case that `bconv_input_op.pad_values()` is not `0` or `1` (which should never happen if the rest of the code does the right thing) it would pass as well. So checking agains ` != 1` should be more conservative.

Reply:
Ah sorry I misread the suggestion :)